### ISSUE TITLE
Abort sync if dest path root does not exists, like rsync

### ISSUE THREAD

**Issue by momiji** [rclone/rclone#5779]
#### What is your current rclone version (output from `rclone version`)?

rclone v1.57.0
- os/version: unknown
- os/kernel: 4.4.59+ (x86_64)
- os/type: linux
- os/arch: amd64
- go/version: go1.17.2
- go/linking: static
- go/tags: none

#### What problem are you are trying to solve?

As per documentation https://rclone.org/commands/rclone_sync/:

> If dest:path doesn't exist, it is created and the source:path contents go there.

However, when using mounted folders (expecially on linux world), it is a common practive to always use a subfolder of mounted path when syncing files to ensure mount is working, hence preventing copying files on local file system instead of remote system.

For example, if mounting a remote folder (whatever protocol, smb, cifs, nfs, s3, ...) in /mnt/data, you should always use /mnt/data/FOLDER/ as source or destination as if /mnt/data is not mounted, rsync from/to /mnt/data/FOLDER/... will fail as FOLDER does not exist and will not be created by rsync as it assumes it should exist.

In rclone, FOLDER will be created whatever, leading to a non-empty /mnt/data folder, which is definitely what is wanted.

#### How do you think rclone should be changed to solve that?

I understand it is very too late to change the default behavior of rclone, however I suggest a new option can be added to prevent creation of root destination folders, which may work on backends that can support it (like local file systems, but not for s3 as there is directory thing).

Maybe an option like --create-remote that allows or prevent creation of remote root path.

I saw an issue about mkdirsAll, and what I'm trying to explain is that mkdirs should be done from root folder to prevent root folder creation, any time it is used. Which will also solve the issue if the mount is broken for whatever reason during the copy.

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by animosity22** [comment]
The problem though is the expectation is that it will create the destination folder as it always has so folks have coded work flows and use it.

A change like that would break any existing workflow and be huge imo.

**Comment by ncw** [comment]
Rclone would have to do a check to see if the destination exists first as everything in rclone creates directories on the fly.

So a flag would need to be something like `--check-destination-exists` or something like that.

How does that sound?

**Comment by momiji** [comment]
@animosity22, this is right, we need to be sure to not change the default behavior everybody is used to.

The idea was to have the default value to create folders, that may be changed for more experienced users who are used to rsync.

So if we go for a `--create-remote` flag, default value should be `true`.
And if we go for `--check-destination-exists`, default value  should be `false`.

I don't know what is better, as I'm not so fluent in english, but I'm sure you've got the point ;)

**Comment by momiji** [comment]
Hi, I find another way to do it on linux, which is widely used for other similar issues for a lot of programs and syncing solutions.

To prevent sofwatre from creating folder even when deep creation is used (like `mkdir -p`), the solution is to add a linked folder in the path.

```
# mount a reomte folder, it may succeed, or not
$ mount //remote/folder /mnt/test

# on normal usage, if we try to write a file in a mounted path, it will always succeed, even if mount failed, as /mnt/test is an empty directory on the file system. The only security here is that the next mount may fail if an option is there to prevent mounts o non-empty folder, I'm not sure if this is possible.
$ touch /mnt/test/file
# the same on a subfolder
$ mkdir /mnt/test/sub/
$ touch /mnt/test/sub/file

# a more secured way for doing this will be to use a folder link. In that case, file creation will definetly fail if the folder is not mounted, at the condition that the link goes to a folder below the mounted path, like this.
# here, file creation will definetly fail if the mount did not succeed :)
$ ln -s /mnt/test/sub /mnt/test-sub-link
$ touch /mnt/test/sub/file
```
Maybe adding a small comment in the documentation explaining with better words what I tried to explain here is enough to close and cancel this feature?
